models2/Preferences.es6.js

```diff
@@ -0,0 +1,118 @@
+import RedditModel from './RedditModel';
+const T = RedditModel.Types;
+
+export default class PreferencesModel extends RedditModel {
+  static type = 'preferences';
+
+  static PROPERTIES = {
+    affiliateLinks: T.bool,
+    allowClicktracking: T.bool,
+    beta: T.bool,
+    clickgadget: T.bool,
+    collapseReadMessages: T.bool,
+    compress: T.bool,
+    credditAutorenew: T.bool,
+    defaultCommentSort: T.string, // It would be nice to have T.oneOf like react here
+    // as it's one of 'confidence', 'old', 'top', 'qa', 'controversial', 'new',
+    defaultThemeSr: T.string,
+    domainDetails: T.bool,
+    emailMessages: T.bool,
+    enableDefaultThemes: T.bool,
+    hideAds: T.bool,
+    hideDowns: T.bool,
+    hideFromRobots: T.bool,
+    hideLocationbar: T.bool,
+    hideUps: T.bool,
+    highlightControversial: T.bool,
+    highlightNewComments: T.bool,
+    ignoreSuggestedSort: T.bool,
+    labelNsfw: T.bool,
+    lang: T.string,
+    legacySearch: T.bool,
+    markMessagesRead: T.bool,
+    media: T.string, // Another case for T.oneOf,
+    // 'on', 'off', 'subreddit'
+    minCommentScore: T.number, // T.number should maybe have a T.range version
```

Comment:
same question as with typeOf. we don't want to throw an error (i don't think. i dunno, maybe we do?), but what would this value be if it doesn't validate? NaN?